### PR TITLE
76 create platform independent post build event for copying dlls

### LINKED ISSUE #76: Create platform independent post-build event for copying dlls
The Linux build is broken until this happens. You cannot use xcopy or copy on Ubuntu

### LINKED ISSUE #77: Clean up `makefile`

### LINKED ISSUE #78: Clean up `rSharp.vcxproj`
There are still tons of configurations with x86 and mono and and and...

### LINKED ISSUE #80: Clean up rSharp.props.in

### LINKED ISSUE #82: Are these files used?
- [ ] create_lib.cmd
- [ ] get_monodsk.cmd
- [ ] get_msbuildpath.cmd
- [ ] get_winsdk.cmd
- [ ] makefile
- [ ] Makevars.in
- [ ] mono.def
- [ ] R.def
- [ ] rSharp-mono.def
- [ ] rSharp-win.def
- [ ] setup_vcpp.cmd
- [ ] win_cp.cmd

### PR DESCRIPTION
**Pull request by rwmcintosh**
Fixes #76
Fixes #78
Fixes #82
Fixes #80
Fixes #77

### CODE REVIEW

File: R/rSharp-env.R [hunk 1]
```diff
@@ -11,7 +11,7 @@ if (.Platform$OS.type == "windows") {
   rSharpEnv$nativePkgName <- "RsharpMs"
 } else {
   # Assuming that everything non-windows is Linux (or will simply not work at all)
-  rSharpEnv$nativePkgName <- "RsharpUX"
+  rSharpEnv$nativePkgName <- "rSharpUX"
```
Comment (rwmcintosh) [inline R/rSharp-env.R:14]: that has to match the case for Linux

File: shared/ClrFacade/ClrFacade.csproj [hunk 2]
```diff
@@ -26,6 +26,13 @@
   <PackageReference Include="R.NET" Version="1.8.1" />
   </ItemGroup>
   <Target Name="PostBuild" AfterTargets="PostBuildEvent">
-    <Exec Command="xcopy /Y $(TargetDir)*.dll $(SolutionDir)..\inst\lib\&#xD;&#xA;xcopy /Y $(SolutionDir)RSharp.runtimeconfig.json $(SolutionDir)..\inst\lib\" />
+    <ItemGroup>
+      <AssemblyFiles Include="$(TargetDir)*.dll" />
+      <ConfigFiles Include="$(SolutionDir)*.runtimeconfig.json" />
+    </ItemGroup>
+    <Copy
```
Comment (rwmcintosh) [inline shared/ClrFacade/ClrFacade.csproj:33]: Nice! this appears to not be very widely used. Most people are doing xcopy and such and that mostly works for windows only solutions.

File: shared/makefile [hunk 3]
```diff
@@ -124,12 +58,11 @@ rSharpInstrSharpUX: $(OUT_DIR)/rSharpUX.so
...
 rSharpLib: rSharpLibComp
 	if [ -e symbols.rds ] ; then cp -u -p symbols.rds $(INSTDIR)/lib$(R_ARCH) ; fi
-	-rm -f $(INSTDIR)/lib/*.*
-	-cp -u -p $(CLR_FACADE_BINS) $(INSTDIR)/lib
+
 	-cp -u -p ./$(RUNTIMECONFIG).json $(INSTDIR)/lib/
 
 rSharpNugetRestore:
-	-$(DOTNET_CMD) restore
+	-$(NUGET_CMD) restore
```
Comment (rwmcintosh) [inline shared/makefile:65]: This is a bit of a pain, I'll open an issue for it. For the time being, we need to install nuget on Ubuntu to restore packages because we are using a file called ```packages.config``` to force restoring dependencies for native C++ projects. That is not supported by ```dotnet``` cli. 

Using ```nuget``` means we need to have mono installed on the build computer.

File: shared/rSharp.cpp [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 using string_t = std::basic_string<char_t>;
 
-const wchar_t* dotnetlib_path = nullptr;
+const char_t* dotnetlib_path = nullptr;
```
Comment (rwmcintosh) [inline shared/rSharp.cpp:12]: @Ian-Du I was all wrong about this when we chatted yesterday. There's a difference in character width between platforms. That's driven by nethost libraries. So, common code needs to use this char_t.

When writing platform specific code within #ifdef WINDOWS I switched to using the explicit type.

File: tests/testthat/datetime-functions.r [hunk 5]
```diff
@@ -129,7 +129,6 @@ post1971_DateStr <- c(
   "2013-10-06 01:59", # DST starts in Canberra
   "2013-10-06 03:01",
   "2013-04-07 01:59", # DST ends in Canberra. See also notes below for peculiarities over the twice run over 02:00 to 03:00
-  "2013-04-07 02:33",
```
Comment (rwmcintosh) [inline tests/testthat/datetime-functions.r:132]: that's an edge case that only passes on windows

**Review by PavelBal — APPROVED** [review]
